mvvm-test/src/main/java/com/thefuntasty/mvvm/test/BaseViewModelExtensionsMock.kt

```diff
@@ -0,0 +1,75 @@
+package com.thefuntasty.mvvm.test
+
+import androidx.lifecycle.LiveData
+import com.thefuntasty.mvvm.BaseViewModel
+import com.thefuntasty.mvvm.livedata.DefaultValueLiveData
+import com.thefuntasty.mvvm.livedata.DefaultValueMediatorLiveData
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.invoke
+
+/**
+ * Extension function that helps with mocking of [BaseViewModel.observeWithoutOwner]
+ *
+ * @return Captured lambda that is passed to original method. [invoke] should be called in order to simulate view state change.
+ *
+ * Usage:
+ *
+ * // GIVEN
+ * val capturedObserveLambda = viewModel.everyObserveWithoutOwner { viewState.number }
+ *
+ * // WHEN
+ * capturedObserveLambda.invoke(...)
+ *
+ */
+fun <VALUE> BaseViewModel<*>.everyObserveWithoutOwner(liveData: () -> LiveData<VALUE>): (VALUE) -> Unit {
```

Comment:
1. Why have you defined 1st param as type `liveData: () -> LiveData<VALUE>` and not `liveData: LiveData<VALUE>`?
2. Method name `everyObserveWithoutOwner` doesn't say what method does nor what it returns in my opinion. What about `mockObserveWithoutOwnerAnswer`, `mockObserveWithoutOwnerLambda` or something similar?

Reply:
1. because when it is passed as an argument to `every{}` mockk is not able to mock that for some reason 🤷‍♂ 
2. I was trying to keep naming similar to mockk. But I will change it to `mockObserveWithoutOwner` It sounds better to me